src/nvim/event/socket.c

```diff
@@ -45,6 +45,12 @@ int socket_watcher_init(Loop *loop, SocketWatcher *watcher,
       return UV_EINVAL;
     }
 
+    if (*port == NUL) {
```

Comment:
A port that looks like a number is used directly, otherwise a lookup is performed.  We should be fine here.